File: resources/exercises/maze/randommouse/RandomMouseMazeEntity.py
import java.util.Random
rand = java.util.Random()

def random3():
    n = rand.nextInt(3)
    if n<0.33:
        return 0
    elif n <0.66:
        return 1
    return 2

def setX(i):
        errorMsg("Sorry Dave, I cannot let you use setX(x) in this exercise. Walk to your goal instead.")
def setY(i):
        errorMsg("Sorry Dave, I cannot let you use setY(y) in this exercise. Walk to your goal instead.")
def setPos(x,y):
        errorMsg("Sorry Dave, I cannot let you use setPos(x,y) in this exercise. Walk to your goal instead.")

# BEGIN SOLUTION
while not isOverBaggle():
    n = random3()
    if n == 0:
        if not isFacingWall():
            forward()
    elif n == 1:
        left()
    else:
        right()
pickupBaggle()
# END SOLUTION

File: resources/exercises/maze/randommouse/RandomMouseMazeEntity.c
//RemoteBuggle

int random3() {
	return rand()%3;
}
/* BEGIN TEMPLATE */
void run() {
	// Your code here =)
	/* BEGIN SOLUTION */
	while (!isOverBaggle()) {
		switch(random3()) {
		case 0:
			if (!isFacingWall()){
				forward(1);
			}
			break;
		case 1:
			left();
			break;
		case 2:
			right();
			break;
		}
	}
	pickupBaggle();
	/* END SOLUTION */
}
/* END TEMPLATE */

